### PR TITLE
api docker: Fix unhandled exception error in git_version

### PR DESCRIPTION
**Pull request by katsel**
#181 describes that the git version cannot be determined because of file system permissions on the `.git/` directory (when using the default docker-compose setup).

To remedy this,

- www-data is granted ownership of all of /opt/cnaas/venv/ (this includes the .git directory, and allows package installs via pip as www-data)
- the cnaas-setup-branch script is run as www-data, not root (to avoid messing up permissions)
- the venv is activated when installing requirements via cnaas-setup-branch script (to avoid installing the requirements into the user's homedir)

### CODE REVIEW

**Comment by katsel** [comment]
Nb. I tried the patch on our lab instance. Docker compose build finished without errors, the API container appears fully functional, and the git_version number is displayed correctly when querying the API.

It may be worth noting that the login user for the API container (via docker exec) changed to www-data, which allows manually switching the branch without worrying about permissions.

File: docker/api/Dockerfile [hunk 1]
```diff
@@ -46,10 +46,12 @@ RUN chown -R www-data:www-data /tmp/devicecerts && \
 # Give permission for unittests
 RUN chown root:www-data /opt/cnaas/*.sh && \
     chmod g+rx /opt/cnaas/*.sh
-RUN chown -R www-data:www-data /opt/cnaas/venv/cnaas-nms/src/
+RUN chown -R www-data:www-data /opt/cnaas/venv/
```
Comment (indy-independence) [inline docker/api/Dockerfile:49]: Why is this needed?

File: docker/api/Dockerfile [hunk 2]
```diff
@@ -46,10 +46,12 @@ RUN chown -R www-data:www-data /tmp/devicecerts && \
...
 
 # Branch specific, don't cache
 ADD "https://api.github.com/repos/SUNET/cnaas-nms/git/refs/heads/" latest_commit
+
+USER www-data
```
Comment (indy-independence) [inline docker/api/Dockerfile:54]: This is what actually solves the problem right?

**Review by indy-independence — CHANGES_REQUESTED** [review]

File: docker/api/Dockerfile (hunk 1 above)
Comment (katsel) [reply docker/api/Dockerfile:49]: `cnaas-setup-branch.sh` does two things:

1. checkout the branch
2. install requirements via pip

For 1., it needs permissions on .git.
For 2., it needs permissions on the venv.

If www-data cannot write to the venv, execution of the script fails and `docker-compose build` exits with an error.

File: docker/api/Dockerfile (hunk 2 above)
Comment (katsel) [reply docker/api/Dockerfile:54]: After some hours of vigorous testing and multiple docker-compose build runs, I am convinced that for this patch to work, all changes are needed.

- user must be www-data
- venv must be activated when running pip install in `cnaas-setup-branch.sh`. Otherwise, requirements are installed in ~SOME PLACE~ (= the user's home dir) and never used by uwsgi.
- for this, www-data needs write access on the venv

It's the most minimally invasive quickfix for #181 that I can come up with right now, without refactoring the whole API Dockerfile (which is another task … for another day).

File: docker/api/Dockerfile (hunk 1 above)
Comment (indy-independence) [reply docker/api/Dockerfile:49]: Ah ok that makes sense. But in this case I think the cnaas-setup-branch.sh should be run as root, and then we change permissions after so it matches what we do for non-branch builds.
I think this is the relevant chown things we do in the main docker thing:
chown -R root:www-data /opt/cnaas/venv/cnaas/
chmod -R u=rwX,g=rX,o= /opt/cnaas/venv/cnaas/
chown -R www-data:www-data /opt/cnaas/venv/cnaas-nms/src/
Ideally the last thing where we change owner of src/ to www-data should not be needed, it says in the comment it's for unit tests but I can't remember why it's needed. Maybe just for creating the .coverage files

Comment (indy-independence) [reply docker/api/Dockerfile:49]: I made a branch to test this out in bugfix.www-data_permissions_branchesandnosetests and also changed so www-data is not permanently the owner of src/ and it seems to work at my end at least

**Comment by katsel** [comment]
I am pulling this PR in favour of just overhauling the whole API Dockerfile.

See #179, where I will add more commits.
